File: Hmwk/terminalVelocity/terminalVelocity.py
def sqrt (value):
	xGuess = value/2.0
	xNext = None
	xPrev = xGuess
	#another previous term added for a value that jumps back and forth
	xPrevTwo = xGuess
	while xNext != xPrev and xNext != xPrevTwo:
		xNext = xGuess - (xGuess*xGuess-value)/(2*xGuess)
		xPrevTwo = xPrev
		xPrev = xGuess
		xGuess = xNext
	return xNext

if __name__=="__main__":
	print("What is the weight of the object?")
	weight = float(input())
	print("What is the coefficient of drag of the object?")
	coefficient = float(input())
	print("What is the area of the object?")
	area = (float(input())
	print("What is the density of the object?")
	density = float(input())
	velocity = sqrt((2*weight)/(coefficient*density*area))
	print("terminal velocity is "+str(velocity))
	dynamicPressure = 0.5*density*velocity**2
	print("Dynamic pressure at terminal velocity is "+str(dynamicPressure))

File: Hmwk/terminalVelocity/terminalVelocity.c
#include<stdio.h>
#include<math.h>

#define PI 3.1415926535897932

/* For problem:
 * diameter = 0.14 ft
 * weight = .10125 lbs
 * coefficient of drag = .5
 * density of air = .0023679
 */

int main () {
	float diameter = 0.14;
	float weight = 0.10125;
	float coefficient = 0.5;
	float density = 0.0023679;
	float terminalVelocity;
	float dynamicPressure;
	float area;

	area = (diameter*diameter*PI)/4.0f;
	terminalVelocity = sqrt((2.0f*weight)/(density*area*coefficient));
	printf("Final velocity is %f\n",terminalVelocity);
	dynamicPressure = 0.5*density*terminalVelocity*terminalVelocity;
	printf("Dynamic pressure is %f\n",dynamicPressure);

	return 0;
}
